src/pybamm/solvers/idaklu_solver.py

```diff
@@ -984,15 +984,14 @@ def _rhs_dot_consistent_initialization(self, y0, model, time, inputs_dict):
         # calculate the time derivatives of the differential equations
         input_eval = inputs if casadi_format else inputs_dict
 
-        rhs_alg0 = model.rhs_algebraic_eval(time, y0, input_eval)
-        if isinstance(rhs_alg0, casadi.DM):
-            rhs_alg0 = rhs_alg0.full()
-        rhs_alg0 = rhs_alg0.flatten()
-
-        rhs0 = rhs_alg0[: model.len_rhs]
+        rhs0 = model.rhs_eval(time, y0, input_eval)
+        if isinstance(rhs0, casadi.DM):
+            rhs0 = rhs0.full()
+        rhs0 = rhs0.flatten()
 
         # for the differential terms, ydot = M^-1 * (rhs)
-        ydot0[: model.len_rhs] = model.mass_matrix_inv.entries @ rhs0
+        # M^-1 is the identity matrix, so we can just use rhs
```

Comment:
this is true for finite volumes, but not always for finite elements - or do we account for it already somewhere else?

Reply:
Thanks, I didn't know FEM could have different mass matrices -- I can revert this, but that means there are pre-existing bugs in pybammsolvers with FEM. I made an issue https://github.com/pybamm-team/pybammsolvers/issues/31

cc @kratman, we should fix this before making IDAKLU the default